appium-tests/common/common.spec.js

```diff
@@ -0,0 +1,321 @@
+                        nextToRemove = removes.shift();
+                    }
+
+                    function removeNext(item) {
+                        if (typeof item === 'undefined') {
+                            callback();
+                            return;
+                        }
+
+                        if (removes.length > 0) {
+                            nextToRemove = removes.shift();
+                        } else {
+                            nextToRemove = undefined;
+                        }
+
+                        item.remove(function removeSucceeded() {
+                            removeNext(nextToRemove);
+                        }, function removeFailed() {
+                            removeNext(nextToRemove);
+                        });
+                    }
+                    removeNext(nextToRemove);
+                }, callback, obj);
+            }, [])
+            .then(function(result) {
+                if (typeof result !== 'undefined') {
+                    throw result;
+                }
+            });
+    }
+
+    it('contacts.ui.util configuring driver and starting a session', function (done) {
+        getDriver()
+            .fail(fail)
+            .finally(done);
+    }, 5 * MINUTE);
+
+    describe('Picking contacts', function () {
+        afterEach(function (done) {
+            removeTestContacts()
+                .finally(done);
+        }, MINUTE);
+
+        it('contacts.ui.spec.1 Pick a contact', function (done) {
+            driver
+                .then(function () {
+                    return addContact('Test', 'Contact');
+                })
+                .then(function () {
+                    return pickContact('Test Contact');
+                })
+                .then(function (contact) {
+                    expect(contact.name.givenName).toBe('Test');
+                    expect(contact.name.familyName).toBe('Contact');
+                })
+                .fail(saveScreenshotAndFail)
+                .finally(done);
+        }, 5 * MINUTE);
+
+        it('contacts.ui.spec.2 Update an existing contact', function (done) {
+            driver
+                .then(function () {
+                    return addContact('Dooney', 'Evans');
+                })
+                .then(function () {
+                    return renameContact('Dooney Evans', 'Urist', 'McContact');
+                })
+                .then(function (contact) {
+                    expect(contact.name.givenName).toBe('Urist');
+                    expect(contact.name.familyName).toBe('McContact');
+                })
+                .then(function () {
+                    return pickContact('Urist McContact');
+                })
+                .then(function (contact) {
+                    expect(contact.name.givenName).toBe('Urist');
+                    expect(contact.name.familyName).toBe('McContact');
+                })
+                .fail(saveScreenshotAndFail)
```

Comment:
Instead of doing this - you could add a jasmine thing to ensure that on every failure we get screenshot:

https://github.com/webdriverio/webdriverio/issues/556